### DISCUSSION TITLE
Connection establishment time is too long

### DISCUSSION THREAD

**Discussion by ethink** [duckdb/duckdb#3672] [General]
There is a large amount of data and it takes a long time to create a connection
he code for creating the connection is as follows:

Connection conn=DriverManager.getConnection("jdbc:duckdb:E:\test\duckdb\10m.db");

The time required to create a connection is as follows:
Time required to create 10 million data connections: 91 milliseconds
Time required to create 20 million data connections: 170 milliseconds
Time required to create 50 million data connections: 416 milliseconds
Time required to create 100 million data connections: 823 milliseconds

As the amount of data increases, the time required increases significantly.
Is there any way to reduce the creation time?
If you use connection pool, how should you use it?

**Comment by hannes** [comment]
Did you specify a primary key? Try without!
